dagger-functions/src/test/java/io/odpf/dagger/functions/udfs/scalar/ArrayAggregateTest.java

```diff
@@ -132,4 +145,20 @@ public void shouldRegisterGauge() throws Exception {
         arrayAggregate.open(functionContext);
         verify(metricGroup, times(1)).gauge(any(String.class), any(Gauge.class));
     }
+
+    @Test
+    public void shouldResolveInPutTypeStrategyForUnresolvedTypes() {
+        when(callContext.getDataTypeFactory()).thenReturn(dataTypeFactory);
+        InputTypeStrategy inputTypeStrategy = new ArrayAggregate().getTypeInference(dataTypeFactory).getInputTypeStrategy();
+        inputTypeStrategy.inferInputTypes(callContext, true);
+        verify(dataTypeFactory, times(1)).createDataType(any(UnresolvedDataType.class));
+    }
+
+    @Test
+    public void shouldRegisterThreeInputArguments() {
+        when(callContext.getDataTypeFactory()).thenReturn(dataTypeFactory);
+        InputTypeStrategy inputTypeStrategy = new ArrayAggregate().getTypeInference(dataTypeFactory).getInputTypeStrategy();
+        inputTypeStrategy.inferInputTypes(callContext, true);
+        Assert.assertEquals(new Integer(3), inputTypeStrategy.getArgumentCount().getMaxCount().get());
```

Comment:
1. assertion suggestion 
"assertEquals(ConstantArgumentCount.of(3), inputTypeStrategy.getArgumentCount());"
2. suggest to use static import for assertEquals